pkg/collector/corechecks/cluster/ksm/kubernetes_state_label_joins.go

```diff
@@ -123,6 +125,28 @@ func newLeafNode() *node {
 	}
 }
 
+// resolveTag resolves the tag key for a given label name and config.
+// It substitutes the %%label%% or %%annotation%% placeholder with the label name.
+// If the label isn't a "matching label", it will be prefixed by either "label_" or "annotation_",
+// if the label is NOT a "matching label", so we can use that to determine which substitution to do.
+func resolveTag(labelName string, config *joinsConfig) string {
+	var ddTagKey string
+	if strings.HasPrefix(labelName, "label_") {
```

Comment:
nit: Following should be cleaner since `ddTagKey` is meant for returning only:

```go
	if strings.HasPrefix(labelName, "label_") {
		return strings.ReplaceAll(
			config.wildcardTemplate,
			"%%label%%",
			strings.TrimPrefix(labelName, "label_"),
		)
	} 
    return strings.ReplaceAll(
			config.wildcardTemplate,
			"%%annotation%%",
			strings.TrimPrefix(labelName, "annotation_"),
		)
```